### ISSUE TITLE
Link element // No title displayed in admin preview.

### ISSUE THREAD

**Issue by DavidFeller** [sulu/sulu#6525]
| Q | A
| --- | ---
| Sulu Version | 2.4.1
| PHP Version | 8.1
| DB Version | MariaDB 10.7.3
| Browser Version | all

# Actual Behavior

Given an element of content type "link" is used without any parameters or the parameter "enable_title" is set to false. 
When an editor selects a target of type "page" no title is displayed in the element's preview.

![link](https://user-images.githubusercontent.com/2094727/161787033-830946f7-0488-481c-912c-cf3a39ee657f.jpg)

# Expected Behavior

Since this makes it hard for editors to see that a target is set, the target page's title should be displayed in the element's preview.

# Steps to Reproduce

1. Insert a content element with the following configuration to your page.

```

<property name="internal_link" type="link">
    <meta>
        <title lang="en">Link</title>
    </meta>
</property>
```

or

```
<property name="internal_link" type="link" >
    <meta>
       <title lang="en">Link</title>
    </meta>
    <params>
        <param name="enable_title" value="false"/>
    </params>
</property>
```

2. Select any internal page as the link target

# Possible Solutions

see expected behavior.

**Comment by alexander-schranz** [comment]
@DavidFeller Thx for report. This should as I think normally return the title of the LinkProvider: https://github.com/sulu/sulu/blob/9ffae24890c56898be8a44d70e796377e6d395dd/src/Sulu/Bundle/MarkupBundle/Markup/Link/LinkItem.php#L68

This already seems correclty be returned and working in the text editor:

<img src="https://user-images.githubusercontent.com/1698337/161796498-a21cab35-3540-4fb4-bcce-3783b796154d.gif" width="320">

So I think we need todo the same as we are doing in the texteditor for the displayed title in the link field type. If there is no title inserted or no title field it should fallback to the link provider title. (The link provider title should not be saved as it is just here to display as if I change the targets title it should also shown that title).

If you want to give it a try to contribute a fix for this, I think somewhere here is maybe the error: https://github.com/sulu/sulu/blob/9ffae24890c56898be8a44d70e796377e6d395dd/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/Link.js#L90, but not sure.

**Comment by DavidFeller** [comment]
@alexander-schranz I did some testing and I think I was able to fix it locally. 

In https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/Link.js#L118 title is always set to undefined if enableTitle is not true. In my case enableTitle is set to false, but nevertheless a title exists up to this point which then is overridden with undefined. This shouldn't happen if the LinkProvider was able to get a title for the target.

```
// Original 
    changeValue = (provider: ?string, href: ?string | number, title: ?string, target: ?string, anchor: ?string) => {
        const { onChange, onFinish, enableTarget, enableTitle, enableAnchor, locale } = this.props;

        onChange({
            provider,
            target: enableTarget ? target : undefined,
            anchor: enableAnchor ? anchor : undefined,
            href,
            title: enableTitle ? title : undefined,
            locale: toJS(locale),
        });
        onFinish();
    };````

```
    // Fix
    changeValue = (provider: ?string, href: ?string | number, title: ?string, target: ?string, anchor: ?string) => {
        const { onChange, onFinish, enableTarget, enableTitle, enableAnchor, locale } = this.props;

        onChange({
            provider,
            target: enableTarget ? target : undefined,
            anchor: enableAnchor ? anchor : undefined,
            href,
            title: enableTitle || title.length ? title : undefined,
            locale: toJS(locale),
        });
        onFinish();
    };```

Since I don't have any experience with jest I won't be able to write sufficient tests. Would it be ok if I created a PR and you guys add tests if needed? ;-)

**Comment by alexander-schranz** [comment]
This was fixed in https://github.com/sulu/sulu/pull/6648.

**Comment by niklasnatter** [comment]
For completion: it looks like displaying the title for `external` links was fixed in #6675 and therefore works on 2.5.* versions only at the moment 🙂
